app/controllers/registration_controller.rb

```diff
@@ -34,24 +24,37 @@ def workflow_list
     end
 
     respond_to do |format|
-      format.any(:json, :xml) { render request.format.to_sym => result.flatten.sort }
+      format.any(:json, :xml) { render request.format.to_sym => result.flatten.uniq.sort }
     end
   end
 
+  ##
+  # Data route to return all the registration collections listed for the given APO
+  # @option params [String] `:apo_id` the druid for the APO
+  # @option params [String] `:truncate` word boundary truncation limit
   def collection_list
-    res = {'' => 'None'}
-    apo_object = Dor.find(params[:apo_id], :lightweight => true)
-    adm_xml = apo_object.administrativeMetadata.ng_xml
-    adm_xml.search('//registration/collection').each do |col|
-      solr_doc = Blacklight.solr.find({:q => "id:\"#{col['id']}\"", :rows => 1, :fl => 'id,tag_ssim,dc_title_tesim'}).docs
-      if solr_doc.first['dc_title_tesim'] && solr_doc.first['dc_title_tesim'].first
-        res[col['id']] = "#{solr_doc.first['dc_title_tesim'].first}(#{col['id']})"
+    truncate_limit = (params[:truncate] || 60).to_i
+    collections = { '' => 'None' }
+    apo_object = Dor.find(params[:apo_id])
+    apo_object.administrativeMetadata.ng_xml.search('//registration/collection/@id').each do |node|
+      col_id = node.to_s
+      col_druid = col_id.gsub(/^druid:/, '')
+      col_title_field = :dc_title_tesim # TODO: why is this :dc_title_tesim and not :dc_title_ssi?
```

Comment:
this actually probably should be `dc_title_ssi`.  both should exist.  `_tesim`'s there for stemmed search, `_ssi`'s there for faceting and exact search and display.  i think i tended to default to `_tesim` when in doubt when i was converting fields early on, so it's possible that i just picked the wrong variant for that or some other reason (without looking back through commit history, i'm assuming this was one of the many field names i updated earlier this year).